### DISCUSSION TITLE
1.7.0 deconstruct toolpath

### DISCUSSION THREAD

**Discussion by lin-ycv** [visose/Robots#274] [Q&A] [answered]
What's this new component, when is it used and how?

**Comment by visose** [comment, marked as answer]
The Create Program component has an input for a Toolpath parameter. This can be fed a list of targets or a single Toolpath object that internally contains a list of targets. 

The Robots plugin by itself doesn't contain components that create Toolpath objects. You can do this with scripting components or with other plugins, like the Extensions plugin.

Some users need to tweak these Toolpath objects after being created, by deconstructing them to a list of targets. 

They use a scripting component for this, but due to some inefficiencies on how scripting components work, for very long lists of targets it can take many seconds, while a proper component is instant.
